webhook/resources/backup/mutator.go

```diff
@@ -77,6 +78,24 @@ func (b *backupMutator) Create(request *admission.Request, newObj runtime.Object
 	}
 	patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/labels", "value": %s}`, string(valueBackupLabels)))
 
+	if backup.Spec.BackupTargetURL == "" {
+		backupTarget, err := b.ds.GetBackupTargetRO(backup.Spec.BackupTargetName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, werror.NewInvalidError(errors.Wrapf(err, "failed to get backup target of backup").Error(), "")
+			}
+			backupTarget, err = b.ds.GetDefaultBackupTargetRO()
+			if err != nil {
+				return nil, werror.NewInvalidError(errors.Wrapf(err, "failed to get default backup target").Error(), "")
+			}
+		}
+		backupTargetURL, err := json.Marshal(backupTarget.Spec.BackupTargetURL)
```

Comment:
So if we can't find the backup target by name, we will use the default backup target URL, but leave the backup's target name field intact?  So the backup's target name and target URL might not match?

Reply:
Added a patchOp to patch the backupTargetURL. Thanks